File: backend/app/utils/phone.py
import re

PY_MOBILE_REGEX = re.compile(r"^09\d{8}$")


def digits_only(value: str) -> str:
    return "".join(c for c in value if c.isdigit())


def normalize_paraguay_phone(value: str | None) -> str | None:
    """Guarda/valida formato local: 0961732207 (10 dígitos)."""
    if not value:
        return None
    digits = digits_only(value)
    if not digits:
        return None
    if PY_MOBILE_REGEX.match(digits):
        return digits
    raise ValueError("Teléfono inválido. Use formato paraguayo: 09XXXXXXXX (10 dígitos)")


def to_whatsapp_phone(value: str) -> str:
    """Convierte 0961732207 → 595961732207 para wa.me."""
    digits = digits_only(value)
    if digits.startswith("595") and len(digits) == 12:
        return digits
    if PY_MOBILE_REGEX.match(digits):
        return f"595{digits[1:]}"
    if len(digits) == 9 and digits.startswith("9"):
        return f"595{digits}"
    raise ValueError("Teléfono inválido para Paraguay")

File: frontend/src/utils/phone.js
/** Teléfono móvil Paraguay: 10 dígitos, formato 09XXXXXXXX */

const PY_MOBILE_REGEX = /^09\d{8}$/

export function digitsOnly(value) {
  return (value || '').replace(/\D/g, '')
}

/** Limita y formatea mientras el usuario escribe (solo dígitos, máx. 10). */
export function formatParaguayPhoneInput(value) {
  let digits = digitsOnly(value).slice(0, 10)
  if (digits.length > 0 && digits[0] !== '0') {
    digits = `0${digits}`.slice(0, 10)
  }
  if (digits.length > 1 && digits[1] !== '9') {
    digits = '09' + digits.slice(2)
  }
  if (digits.length === 1 && digits !== '0') {
    digits = '0'
  }
  return digits
}

export function isValidParaguayPhone(value) {
  if (!value) return true
  return PY_MOBILE_REGEX.test(digitsOnly(value))
}

/** Normaliza para guardar en BD: 0961732207 */
export function normalizeParaguayPhone(value) {
  const digits = digitsOnly(value)
  if (!digits) return null
  if (PY_MOBILE_REGEX.test(digits)) return digits
  return null
}

/** Muestra con espacios: 0961 732 207 */
export function displayParaguayPhone(value) {
  const digits = digitsOnly(value)
  if (!digits) return ''
  if (digits.length <= 4) return digits
  if (digits.length <= 7) return `${digits.slice(0, 4)} ${digits.slice(4)}`
  return `${digits.slice(0, 4)} ${digits.slice(4, 7)} ${digits.slice(7, 10)}`
}
